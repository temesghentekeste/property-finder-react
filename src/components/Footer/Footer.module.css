.mainFooter {
  height: 40vh;
  margin-top: auto;
  align-self: flex-end;
  padding: 1.5rem 0.4rem 5rem;
  background: #9e9a9a81;
  text-align: center;
  color: rgba(12, 12, 12, 0.911);
  font-size: 0.85rem;
  opacity: 1;
}

.mainFooter__text {
  margin-top: auto;
  display: flex;
  justify-content: center;
  align-items: flex-end;
  color: rgb(15, 15, 15);
  font-size: 0.85rem;
  font-weight: bold;
}

footer a {
  background: none;
  display: inline-block;
  color: rgb(7, 7, 7);
}

.mainFooter a {
  transition: ease-in-out 0.3s;
}
