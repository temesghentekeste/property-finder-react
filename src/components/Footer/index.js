import styles from './Footer.module.css';

const Footer = () => (
  <footer className={styles.mainFooter}>
    <p className={styles.mainFooter__text}>
      <a href="https://www.temesghen.me" target="_blank" rel="noreferrer">
        &copy;Temesghen Tekeste
      </a>
      2021, All Rights Reserved
    </p>
  </footer>
);

export default Footer;
