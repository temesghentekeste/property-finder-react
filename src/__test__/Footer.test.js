import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Footer from '../components/Footer';

let getByTestId;

beforeEach(() => {
  const component = render(<Footer />);
  getByTestId = component.getByTestId;
});

it('renders the Footer component', () => {
  const component = renderer.create(<Footer />).toJSON();

  expect(component).toMatchSnapshot();
});

test('should render Footer component with correct text', () => {
  const footerEl = getByTestId('footer');

  expect(footerEl.textContent.includes('Temesghen Tekeste')).toBe(true);
});
