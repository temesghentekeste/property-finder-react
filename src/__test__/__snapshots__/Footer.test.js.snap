// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Footer component 1`] = `
<footer
  className="mainFooter"
  data-testid="footer"
>
  <p
    className="mainFooter__text"
  >
    <a
      href="https://www.temesghen.me"
      rel="noreferrer"
      target="_blank"
    >
      ©Temesghen Tekeste
    </a>
     2021, All Rights Reserved
  </p>
</footer>
`;
